### ISSUE TITLE
Error when listing collections - tries to use local directory as self href

### ISSUE THREAD

**Issue by robintw** [stac-utils/pystac-client#113]
I've got a weird error that I've been trying to debug, but have got a bit lost in the pystac and pystac-client code.

I'm trying to use the CLI to get a list of collections from a STAC API. The API is being served by a stac-fastapi server.

When running:

```
stac-client collections http://localhost:8000
```

I get an error of
```
No connection adapters were found for 'C:\\Users\\rwilson3\\Documents\\code/collections'
```

It seems to be creating a collections URL that is the current directory plus `/collections`.

Looking at the code, it seems like the self href for the catalog is being returned wrongly. I've checked the raw JSON coming from `http://localhost:8000`, and it lists a correct self href of:

```
{
"rel": "self",
"type": "application/json",
"href": "http://localhost:8000/"
}
```

Also, the URL `http://localhost:8000/collections` returns the correct collections JSON.

Am I doing something wrong here? Or is stac-fastapi possibly doing something wrong?

This all works fine for the example given in the docs (`https://earth-search.aws.element84.com/v0`) - but I can't see what is different about that URL: it seems to give the same sort of result for the self relation.

In case it helps for debugging, the full JSON response given when visiting `http://localhost:8000` is given below:
<details>
```
{
"type": "Catalog",
"id": "stac-fastapi",
"title": "stac-fastapi",
"description": "stac-fastapi",
"stac_version": "1.0.0",
"conformsTo": [
"http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/core",
"http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/oas30",
"http://www.opengis.net/spec/ogcapi-features-4/1.0/conf/simpletx",
"https://api.stacspec.org/v1.0.0-beta.3/ogcapi-features",
"https://api.stacspec.org/v1.0.0-beta.3/ogcapi-features/extensions/transaction/",
"https://api.stacspec.org/v1.0.0-beta.3/item-search/#query",
"https://api.stacspec.org/v1.0.0-beta.3/item-search/#fields",
"http://www.opengis.net/spec/ogcapi-features-1/1.0/conf/geojson",
"https://api.stacspec.org/v1.0.0-beta.3/item-search/#context",
"https://api.stacspec.org/v1.0.0-beta.3/item-search/#sort",
"https://api.stacspec.org/v1.0.0-beta.3/item-search",
"https://api.stacspec.org/v1.0.0-beta.3/core"
],
"links": [
{
"rel": "self",
"type": "application/json",
"href": "http://localhost:8000/"
},
{
"rel": "root",
"type": "application/json",
"href": "http://localhost:8000/"
},
{
"rel": "data",
"type": "application/json",
"href": "http://localhost:8000/collections"
},
{
"rel": "docs",
"type": "application/json",
"title": "OpenAPI docs",
"href": "http://localhost:8000/docs"
},
{
"rel": "conformance",
"type": "application/json",
"title": "STAC/WFS3 conformance classes implemented by this server",
"href": "http://localhost:8000/conformance"
},
{
"rel": "search",
"type": "application/geo+json",
"title": "STAC search",
"href": "http://localhost:8000/search",
"method": "GET"
},
{
"rel": "search",
"type": "application/json",
"title": "STAC search",
"href": "http://localhost:8000/search",
"method": "POST"
},
{
"rel": "child",
"type": "application/json",
"title": "joplin",
"href": "http://localhost:8000/collections/joplin"
},
{
"rel": "child",
"type": "application/json",
"title": "test-anglo-azure",
"href": "http://localhost:8000/collections/test-anglo-azure"
},
{
"rel": "service-desc",
"type": "application/vnd.oai.openapi+json;version=3.0",
"title": "OpenAPI service description",
"href": "http://localhost:8000/api"
}
],
"stac_extensions": [
"https://raw.githubusercontent.com/radiantearth/stac-api-spec/v1.0.0-beta.3/fragments/context/json-schema/schema.json"
]
}
```
</details>

**Comment by matthewhanson** [comment]
Sorry for the delay here @robintw , I'm not entirely sure why localhost would be different.  Will give this a try.

**Comment by Samweli** [comment]
I confirm this issue, I have experience the same error when retrieving collections hosted on a locally hosted STAC API.

**Comment by gadomski** [comment]
It looks like this is an upstream issue in **pystac**. I've opened up a PR for a fix here: stac-utils/pystac#697.

**Comment by gadomski** [comment]
This was merged in https://github.com/stac-utils/pystac/commit/63bfe7eb9fdcd21a01b18db67ec397c8dff5f745. It will be included in the next **pystac** release.
